packages/system/src/border.ts

```diff
@@ -1,12 +1,87 @@
 import { BorderKeys } from "./keys";
 import { toCssUnit } from "./toCSS";
-import { CSSProperties, ResponsiveStyle } from "./types";
+import { CSSProperties, CSSValue, ResponsiveStyle } from "./types";
 import { applyResponsiveStyles } from "./responsive";
 
 export type BorderProps = Partial<
-  CSSProperties<"border" | "borderTop" | "borderRight" | "borderLeft" | "borderBottom"> &
-    CSSProperties<"borderStyle"> &
-    CSSProperties<"borderRadius" | "borderWidth", true>
+  {
+    /**
+     * @see borderTop
+     * @see borderBottom
+     */
+    borderY: CSSValue<"borderTop" | "borderBottom", true>;
+    /**
+     * @see borderLeft
+     * @see borderRight
+     */
+    borderX: CSSValue<"borderLeft" | "borderRight", true>;
+    /**
+     * @see borderInlineStart
+     */
+    borderStart: CSSValue<"borderInlineStart", true>;
+    /**
+     * @see borderInlineEnd
+     */
+    borderEnd: CSSValue<"borderInlineEnd", true>;
+    /**
+     * @see borderInlineStartWidth
+     */
+    borderStartWidth: CSSValue<"borderInlineStartWidth", true>;
+    /**
+     * @see borderInlineEndWidth
+     */
+    borderEndWidth: CSSValue<"borderInlineEndWidth", true>;
+    /**
+     * @see borderInlineStartStyle
+     */
+    borderStartStyle: CSSValue<"borderInlineStartStyle">;
+    /**
+     * @see borderInlineEndStyle
+     */
+    borderEndStyle: CSSValue<"borderInlineEndStyle">;
```

Comment:
I don't believe that `borderEndStyle` should accept a numerical value. This concern also extends to some of the other styles added in this PR.

Reply:
@poteboy 
I don't think borderEndStyle accepts numeric values

Reply:
Oh, what I actually meant by that is we should filter the third argument of `applyResponsiveStyles` based on whether or not the property accepts a numerical value.

Reply:
I fixed what you pointed out. 
please confirm.